Add pangeo community collab to peer review guide (and precommit!) (#155)
pyOpenSci/software-peer-review: partners/scientific-communities.md

* A catalog of vetted tools and a community landing page.

### Partnership benefits to the broader scientific community
The benefit of communities partnering with us is:

* We have a broad view across all scientific ecosystems which will help us to better identify overlap of package functions.
* We also have broader reach to users of scientific software across the ecosystem.
* Finally a consolidated peer review process will help enforce standardization of scientific Python packaging and best practices across the ecosystem.

## Pangeo

pyOpenSci is launching a prototype effort to explore what it looks like to
support [Pangeo](https://pangeo.io/).

If you are submitting a package for review that fits into the Pangeo ecosystem,
your package will be:

* Reviewed against current pyOpenSci guidelines and checks.
* Also reviewed against the Pangeo-specific guidelines below.

```{admonition} Pangeo Review Guidelines
:class: note

On top of the pyOpenSci review guidelines, we will also perform checks for
the following:

* **Consume and produce high-level data structures** (e.g. **Xarray datasets**, **Pandas DataFrames**) wherever feasible:

> * **Operate lazily when called on Dask data structure:** Lazy loading refers to a tool only performing operations on data when it's needed. Lazy loading is ideal for workflows using larger datasets which will quickly consume memory and compute power when you work with them. [Learn more here.](https://saturncloud.io/blog/a-data-scientist-s-guide-to-lazy-evaluation-with-dask/)

Author: updated this locally. it's not letting me commit here

pyOpenSci is launching a prototype collaboration that supports an existing
scientific community.

### What pyOpenSci provides to existing communities
In addition to [the core benefits that pyOpenSci peer review provides](/about/benefits),
we also provide infrastructure and resources for:
* Peer review of tools in the community's ecosystem
* Adoption of community specific review standards which will be enforced in addition
 [to our existing review guidelines](editor-checklist-template)
* Maintenance support to maintainers that go through our review.
* Support for finding a new package maintainer in case the current one needs to step down.
* A catalog of vetted tools and a community landing page.

### Partnership benefits to the broader scientific community
The benefit of communities partnering with us is:

* We have a broad view across all scientific ecosystems which will help us to better identify overlap of package functions.
* We also have broader reach to users of scientific software across the ecosystem.
* Finally a consolidated peer review process will help enforce standardization of scientific Python packaging and best practices across the ecosystem.

> ## pyOpenSci's partnership with Pangeo

Author: ```suggestion
(pangeo-collaboration)=
## pyOpenSci's partnership with Pangeo
```

* **Consume and produce high-level data structures** (e.g. **Xarray datasets**, **Pandas DataFrames**) wherever feasible:
* **Operate lazily when called on Dask data structure:** Lazy loading refers to a tool only performing operations on data when it's needed. Lazy loading is ideal for workflows using larger datasets which will quickly consume memory and compute power when you work with them. Lazy loading allows a tool to create a connection to a dataset, but only actually access / open the part of the dataset that you require to use in your workflow.  [Learn more here.](https://docs.dask.org/en/stable/user-interfaces.html?highlight=lazy#laziness-and-computing)

    * A few other resources related to lazy loading:

        * [Dask and stages of computation](https://docs.dask.org/en/stable/phases-of-computation.html): this provides and overview of using Dask effectively to optimize your compute workflow.
        * [Using Dask and Xarray together:](https://docs.xarray.dev/en/stable/user-guide/dask.html?highlight=lazy#using-dask-with-xarray) this page provides an overview of how to use Dask to optimize working with data arrays in Xarray.

> * **Avoid file input and output (I/O) unless your package is specifically an I/O package:** I/0 refers to data input and output of data.

Author: i need some clarification on what this means in practice as it's often useful to write out a file during a workflow as an artifact or for use elsewhere. even xarray i think allows this and also needs to open datasets?

Author: maybe this relates to https://docs.2i2c.org/en/latest/data/cloud.html#cloud-native-formats so rather than simple exporting say txt you want a cloud native format?

Reviewer: Use existing data reading and writing libraries, rather than writing your own. Tools like xarray and it's readers and writers, which build on libraries like Zarr, fsspec, h5netcdf, GDAL, and more can already read and write a variety of file types to a variety of file systems, including Cloud Object storage.

Author: i grabbed a bit of language from your pangeo guide here too

is this the text below ok @TomAugspurger  ?
***************
* **Use existing data reading and writing libraries rather than creating your own read and write utilities:** Unless you package is specifically focused on reading or writing data with specific use cases in mind, it
   should not include its own custom code for read/write data operations. The data be read and written using tools within packages such as Xarray or geoparquet via GeoPandas that build on existing packages such as Zarr, fsspec, h5netcdf, and GDAL. These tools already read and write a variety of file types to a variety of file systems, including Cloud Object storage.
[See: 2i2c documentation for more on cloud native file format support.](https://docs.2i2c.org/en/latest/data/cloud.html#cloud-native-formats)

* **Operate lazily when called on Dask data structure:** Lazy loading refers to a tool only performing operations on data when it's needed. Lazy loading is ideal for workflows using larger datasets which will quickly consume memory and compute power when you work with them. Lazy loading allows a tool to create a connection to a dataset, but only actually access / open the part of the dataset that you require to use in your workflow.  [Learn more here.](https://docs.dask.org/en/stable/user-interfaces.html?highlight=lazy#laziness-and-computing)

    * A few other resources related to lazy loading:

        * [Dask and stages of computation](https://docs.dask.org/en/stable/phases-of-computation.html): this provides and overview of using Dask effectively to optimize your compute workflow.
        * [Using Dask and Xarray together:](https://docs.xarray.dev/en/stable/user-guide/dask.html?highlight=lazy#using-dask-with-xarray) this page provides an overview of how to use Dask to optimize working with data arrays in Xarray.
        * [Tutorial on using Dask and lazy loading]( http://tutorial.dask.org/01_dataframe.html )

> * **Avoid file input and output (I/O) unless your package is specifically an I/O package:** I/0 refers to data input and output of data.

Author: on the cloud - especially important to use cloud native formats - permissions can be challenging and it can be very slow ... 
don't write your own package for reading and writing data - use existing ones like xarray, rasterio, etc...

Pangeo-related package, we will also perform the checks below that are
defined by the Pangeo community.

* **Consume and produce high-level data structures** (e.g. **Xarray datasets**, **Pandas DataFrames**) wherever feasible.
* **Operate lazily when called on Dask data structures:** To support larger-than-memory datasets, many Pangeo workflows use [Dask](https://dask.org) to parallelize the workflow. Doing this efficiently requires lazily loading data and triggering compute only when needed. See the [Dask documentation](https://docs.dask.org/en/stable/user-interfaces.html#laziness-and-computing) for more.

    * A few other resources related to lazy loading:

        * [Dask and stages of computation](https://docs.dask.org/en/stable/phases-of-computation.html): this provides and overview of using Dask effectively to optimize your compute workflow.
        * [Using Dask and Xarray together:](https://docs.xarray.dev/en/stable/user-guide/dask.html?highlight=lazy#using-dask-with-xarray) this page provides an overview of how to use Dask to optimize working with data arrays in Xarray.

>         * [Tutorial on using Dask and lazy loading]( http://tutorial.dask.org/01_dataframe.html )

Author: ```suggestion
        * [Tutorial on using Dask and lazy loading]( https://tutorial.dask.org/01_dataframe.html )
```

> * **Use existing data reading and writing libraries rather than creating your own:** In general, your package should use existing libraries for reading and writing data (e.g. xarray, pandas, geopandas, or the libraries those are built on like Zarr, GDAL, pyarrow, etc.) and should not include its own custom code for read/write data operations. These tools already read and write a variety of file types to a variety of file systems that are important to pangeo users, including Cloud Object storage.

Reviewer 2: ```suggestion
* **Use existing data reading and writing libraries rather than creating your own:** In general, your package should use existing libraries for reading and writing data (e.g. xarray, pandas, geopandas, or the libraries those are built on like Zarr, GDAL, pyarrow, etc.) and should not include its own custom code for read/write data operations. These tools already read and write a variety of file types to a variety of file systems that are important to Pangeo users, including Cloud Object storage.
```

* We have a broad view across all scientific ecosystems which will help us to better identify overlap of package functions.
* We also have broader reach to users of scientific software across the ecosystem.
* Finally a consolidated peer review process will help enforce standardization of scientific Python packaging and best practices across the ecosystem.

(pangeo-collaboration)=
## pyOpenSci's partnership with Pangeo

pyOpenSci is creating a collaboration workflow to
collaborate with the [Pangeo](https://pangeo.io/) scientific community. Through

> this collaboration we add an additional layer of peer review checks to pangeo
> packages that go through our [open peer review process](/about/intro).

Reviewer: ```suggestion
this collaboration we add an additional layer of peer review checks to Pangeo-related
packages that go through our [open peer review process](/about/intro).
```

If you are submitting a package for review that fits into the Pangeo ecosystem,
your package will be:

* Reviewed against current pyOpenSci guidelines and checks.
* Also reviewed against the Pangeo-specific guidelines listed below.

### Pangeo Specific Software Peer Review Guidelines

In addition to [basic pyOpenSci review checks](https://www.pyopensci.org/software-peer-review/how-to/editor-in-chief-guide.html#editor-checklist-template), when reviewing a
Pangeo-related package, we will also perform the checks below that are
defined by the Pangeo community.

* **Consume and produce high-level data structures** (e.g. **Xarray datasets**, **Pandas DataFrames**) wherever feasible.

> * **Operate lazily when called on Dask data structure:** To support larger-than-memory datasets, many Pangeo workflows use [Dask](https://dask.org) to parallelize the workflow. Doing this efficiently requires lazily loading data and triggering compute only when needed. See the [Dask documentation](https://docs.dask.org/en/stable/user-interfaces.html#laziness-and-computing) for more.

Reviewer: ```suggestion
* **Operate lazily when called on Dask data structures:** To support larger-than-memory datasets, many Pangeo workflows use [Dask](https://dask.org) to parallelize the workflow. Doing this efficiently requires lazily loading data and triggering compute only when needed. See the [Dask documentation](https://docs.dask.org/en/stable/user-interfaces.html#laziness-and-computing) for more.
```

    * A few other resources related to lazy loading:

        * [Dask and stages of computation](https://docs.dask.org/en/stable/phases-of-computation.html): this provides and overview of using Dask effectively to optimize your compute workflow.
        * [Using Dask and Xarray together:](https://docs.xarray.dev/en/stable/user-guide/dask.html?highlight=lazy#using-dask-with-xarray) this page provides an overview of how to use Dask to optimize working with data arrays in Xarray.
        * [Tutorial on using Dask and lazy loading]( http://tutorial.dask.org/01_dataframe.html )

> * **Use existing data reading and writing libraries rather than creating your own read and write utilities:** Unless you package is specifically focused on reading or writing data with specific use cases in mind, it
>    should not include its own custom code for read/write data operations. The data be read and written using tools within packages such as Xarray or geoparquet via GeoPandas that build on existing packages such as Zarr, fsspec, h5netcdf, and GDAL. These tools already read and write a variety of file types to a variety of file systems, including Cloud Object storage.
> [See: 2i2c documentation for more on cloud native file format support.](https://docs.2i2c.org/en/latest/data/cloud.html#cloud-native-formats)

Reviewer: ```suggestion
* **Use existing data reading and writing libraries rather than creating your own:** In general, your package should use existing libraries for reading and writing data (e.g. xarray, pandas, geopandas, or the libraries those are built on like Zarr, GDAL, pyarrow, etc.) and should not include its own custom code for read/write data operations. These tools already read and write a variety of file types to a variety of file systems that are important to pangeo users, including Cloud Object storage.
[See: 2i2c documentation for more on cloud native file format support.](https://docs.2i2c.org/en/latest/data/cloud.html#cloud-native-formats)
```
